src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -459,6 +461,11 @@ def set_broker_info(allow_broker):
     _session.allow_broker = allow_broker
 
 
+@decorators.suppress_all_exceptions()
+def set_msal_telemetry(msal_telemetry):
+    _session.msal_telemetry = msal_telemetry
```

Comment:
There is one problem with this code: If a command makes multiple MSAL requests, the last telemetry record will overwrite previous ones. Or, shall we only keep the first telemetry record?

```suggestion
    if not _session.msal_telemetry:
        _session.msal_telemetry = msal_telemetry
```

Reply:
If there can be one telemetry entry per each "az foo ..." command, yet one such command may potentially make multiple MSAL requests, it is indeed unclear how or whether to handle this one-to-many mapping. I would suggest to just pick one MSAL request to collect its telemetry, preferably the one which happened to fail, but any choice is still better than nothing. CC @MSamWils